mongodb/src/main/java/org/hibernate/ogm/datastore/mongodb/configuration/impl/MongoDBConfiguration.java

```diff
@@ -82,14 +86,62 @@ public MongoDBConfiguration(ConfigurationPropertyReader propertyReader, OptionsC
 	 * @return the {@link MongoClientOptions} corresponding to the {@link MongoDBConfiguration}
 	 */
 	public MongoClientOptions buildOptions() {
-		MongoClientOptions.Builder optionsBuilder = new MongoClientOptions.Builder();
+		MongoClientOptions.Builder optionsBuilder = MongoClientOptions.builder();
 
 		optionsBuilder.connectTimeout( timeout );
 		optionsBuilder.writeConcern( writeConcern );
 		optionsBuilder.readPreference( readPreference );
+
+		Map<String, Method> settingsMap = createSettingsMap();
+		for ( Map.Entry<String, Method> entry : settingsMap.entrySet() ) {
+			String setting = MongoDBProperties.MONGO_DRIVER_SETTINGS_PREFIX + "." + entry.getKey();
+			// we know that there is exactly one parameter
+			Class<?> type = entry.getValue().getParameterTypes()[0];
+
+			// for reflection purposes we need to deal with wrapper classes
+			if ( int.class.equals( type ) ) {
+				type = Integer.class;
+			}
+			if ( boolean.class.equals( type ) ) {
+				type = Boolean.class;
+			}
+
+			Object property = propertyReader.property( setting, type ).withDefault( null ).getValue();
+			if ( property == null ) {
+				continue;
+			}
+
+			Method settingMethod = entry.getValue();
+			try {
+				settingMethod.invoke( optionsBuilder, property );
+			}
+			catch ( InvocationTargetException | IllegalAccessException e ) {
+				// TODO OGM-810 fix error message
+			}
+		}
+
 		return optionsBuilder.build();
 	}
 
+	private Map<String, Method> createSettingsMap() {
+		Map<String, Method> settingsMap = new HashMap<>();
+
+		Method[] methods = MongoClientOptions.Builder.class.getDeclaredMethods();
+		for ( Method method : methods ) {
+			if ( method.getParameterTypes().length == 1 ) {
+				Class<?> parameterType = method.getParameterTypes()[0];
+				// we just care of string, int and boolean setters
+				if ( String.class.equals( parameterType )
+						|| int.class.equals( parameterType )
```

Comment:
Actually there is yet another little complication. `WriteConcernType` is not a 1:1 mapping to the possible `WriteConcern` instances.There is also `WriteConcernType#CUSTOM` which maps to no pre-defined `WriteConcern`, but is supposed to be used in conjunction with `hibernate.ogm.mongodb.write_concern_type`. If we were to support the property `hibernate.ogm.mongodb.driver.writeConcern` we would need to drop `WriteConcernType#CUSTOM`, since the user would expect a 1:1 mapping here. 

Personally I find `hibernate.ogm.mongodb.write_concern_type` a bit odd. I think the best solution would be to just have `hibernate.ogm.mongodb.driver.writeConcern` and handle the cases where the value is a string representation of one of the pre-defined write concern instances, a fully qualified class name or an actual `WriteConcern` instance. 

Thougths?